backend/pkg/httpserver/create_saved_search.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpserver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+
+	"github.com/GoogleChrome/webstatus.dev/lib/gcpspanner/searchtypes"
+	"github.com/GoogleChrome/webstatus.dev/lib/gen/openapi/backend"
+	"github.com/GoogleChrome/webstatus.dev/lib/httpmiddlewares"
+)
+
+const (
+	savedSearchNameMaxLength            = 32
+	savedSearchNameMinLength            = 1
+	savedSearchNameDescriptionMaxLength = 1024
+	savedSearchNameDescriptionMinLength = 1
+	savedSearchQueryMaxLength           = 256
```

Comment:
Is there any limit on the number of saved searches that a given user can have?  Some user could try to spam us.

Reply:
We do limit it to 25 per user.
https://github.com/GoogleChrome/webstatus.dev/blob/41442aba41c581fafde65b66a96a005ca37f1c14/lib/gcpspanner/client.go#L127

But that configuration is very hidden. I will make a separate PR to make it obvious during the creation of the spanner [client](https://github.com/GoogleChrome/webstatus.dev/blob/main/backend/cmd/server/main.go#L68)